support/cas-server-support-actions/src/main/java/org/apereo/cas/web/flow/login/mfa/MfaInitializeAction.java

```diff
@@ -0,0 +1,35 @@
+package org.apereo.cas.web.flow.login.mfa;
+
+import org.apereo.cas.services.MultifactorAuthenticationProvider;
+import org.apereo.cas.util.spring.ApplicationContextProvider;
+import org.springframework.webflow.action.AbstractAction;
+import org.springframework.webflow.execution.Event;
+import org.springframework.webflow.execution.RequestContext;
+
+/**
+ * This Action is wired in by AbstractCasMultifactorWebflowConfigurer to put
+ * the resolved provider in the flow scope.
+ *
+ * @author Travis Schmidt
+ * @since 5.3.4
+ */
+public class MfaInitializeAction extends AbstractAction {
```

Comment:
Same as above, really. This probably needs to move into some sort of common MFA module.